pkg/container/SPEC.md

```diff
@@ -0,0 +1,103 @@
+# UCAN container Specification v0.1.0
+
+## Editors
+
+* [Michael Muré], [Consensys]
+
+## Authors
+
+* [Michael Muré], [Consensys]
+* [Hugo Dias]
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [BCP 14] when, and only when, they appear in all capitals, as shown here.
+
+# 0 Abstract
+
+[User-Controlled Authorization Network (UCAN)][UCAN] is a trustless, secure, local-first, user-originated authorization and revocation scheme. This document describes a container format for transmitting one or more UCAN tokens as bytes, regardless of the transport.
+
+# 1 Introduction
+
+The UCAN spec itself is transport agnostic. This specification describes how to transfer one or more [UCAN] tokens bundled together, regardless of the transport.
+
+# 2 Container format
+
+## 2.1 Inner structure
+
+UCAN tokens, regardless of their kind ([Delegation], [Invocation], [Revocation], [Promise]) MUST be first signed and serialized into bytes according to their respective specification. As the token's CID is not part of the serialized container, any CID returned by this operation is to be ignored. 
+
+All the tokens' bytes MUST be assembled in a [CBOR] array, which is then inserted as the value under the `ctn-v1` string key, in a CBOR map. The ordering of tokens in the array MUST NOT matter. For clarity, the CBOR shape is given below:  
+
+```json
+{
+  "ctn-v1": [
```

Comment:
I'm generally a fan of this style, but given the binary representation of everything else in this spec, why not the same here?

`<container-sigil><version><cbor>`

Also what happens if there's more fields in this payload? `MUST`/`SHOULD` they be ignored or is it an extension point?

Reply:
After pondering the question, it boiled down to a very potato/potato choice between:

--> your version:
- more compact
- allow to swap cbor for something else in the future (but really, ucan itself is cbor anyway)
- makes it less obvious that it's a ucan container by just looking at bytes (and thus, might warrant some magic bytes?)

the original `<header byte><cbor bytes, optionally compressed, optionally encoded>`:
- ~a few more~ 7 bytes longer
- where `ctn-v1` is at a fixed position, so more or less act as magic bytes

So, potato/potato, and I'll go for the original proposal as a bit more human friendly.